packages/components/src/components/Button/index.ts

```diff
@@ -7,13 +7,78 @@ export const ButtonComponentDefinition: ComponentDefinition = {
   id: 'button',
   name: 'Button',
   category: 'Contentful',
-  builtInStyles: ['cfMargin', 'cfPadding'],
+  builtInStyles: [
+    'cfMargin',
+    'cfMaxWidth',
+    'cfLetterSpacing',
+    'cfTextItalic',
+    'cfTextUnderline',
+    'cfTextBold',
+    'cfLineHeight',
+    'cfBorder',
+  ],
   thumbnailUrl: constants.thumbnails.button,
   variables: {
+    cfFontSize: {
+      displayName: 'Font Size',
+      type: 'Text',
+      group: 'style',
+      description: 'The font size of the button.',
+      defaultValue: '16px',
+    },
+    cfFontWeight: {
+      displayName: 'Font Weight',
+      type: 'Text',
+      group: 'style',
+      description: 'The font weight of the button.',
+      defaultValue: '600',
+    },
+    cfTextColor: {
+      displayName: 'Text Color',
+      type: 'Color',
+      group: 'style',
+      description: 'The text color of the button.',
+      defaultValue: 'rgba(255, 255, 255, 1)',
+    },
+    cfBackgroundColor: {
+      displayName: 'Background Color',
+      type: 'Color',
+      group: 'style',
+      description: 'The background color of the button.',
+      defaultValue: 'rgba(0, 0, 0, 1)',
+    },
+    cfPadding: {
+      displayName: 'Padding',
+      type: 'Text',
+      group: 'style',
+      description: 'The padding of the button.',
+      defaultValue: '6px 12px 6px 12px',
+    },
+    cfWidth: {
+      displayName: 'Width',
+      type: 'Text',
+      group: 'style',
+      description: 'The width of the button.',
+      defaultValue: '92px',
+    },
+    cfHeight: {
+      displayName: 'Height',
+      type: 'Text',
+      group: 'style',
+      description: 'The height of the button.',
+      defaultValue: '43px',
+    },
```

Comment:
I'm a little worried about the fixed height/width.  Is there a default value along the lines of "auto" that would allow the height and width to be dynamic based on the text content and padding of the button instead?

Reply:
Or maybe if the default size of the button was wider like 200px by 40px that would be nice way to give a little more room for text changes without having to introduce something new to allow an "auto" width option